cla_public/apps/contact/views.py

```diff
@@ -30,31 +33,79 @@ def add_no_cache_headers(response):
         end_time = callback_time + datetime.timedelta(minutes=30)
+        callback_time_string = callback_time.strftime("%A, %d %B at %H:%M - ") + end_time.strftime("%H:%M")
+        return callback_time_string
+
+
+def generate_confirmation_email_data(data):
+    """
+    Generates the data used in the sending of Gov Notify emails;
+    includes paths for 5 different templates based on circumstance
+    """
+    try:
         data.update(
-            {"callback_time_string": callback_time.strftime("%A, %d %B at %H:%M - ") + end_time.strftime("%H:%M")}
+            {
+                "case_ref": session.stored.get("case_ref"),
+                "callback_requested": session.stored.get("callback_requested"),
+                "contact_type": session.stored.get("contact_type"),
+            }
         )
+        email_address = data["email"]
+
+        # Path for confirmation email if no email is provided initially
+        if "full_name" not in data:
+            if session.stored["callback_requested"] is True:
+                personalisation = {"case_reference": data["case_ref"], "date_time": set_callback_time_string(data)}
+                template_id = GOVUK_NOTIFY_TEMPLATES["PUBLIC_CONFIRMATION_EMAIL_CALLBACK_REQUESTED"]
+            else:
+                personalisation = {"case_reference": data["case_ref"]}
+                template_id = GOVUK_NOTIFY_TEMPLATES["PUBLIC_CONFIRMATION_NO_CALLBACK"]
+
+            return email_address, template_id, personalisation
+
+        # Returns email if provided on contact form
+        personalisation = {
+            "full_name": data["full_name"],
+            "thirdparty_full_name": data["thirdparty"]["full_name"],
+            "case_reference": data["case_ref"],
+            "date_time": set_callback_time_string(data),
+        }
 
-    recipient = (data["full_name"], data["email"]) if data.get("full_name") else data["email"]
+        if session.stored["callback_requested"] is False:
+            template_id = GOVUK_NOTIFY_TEMPLATES["PUBLIC_CALLBACK_NOT_REQUESTED"]
 
-    session["confirmation_email"] = data["email"]
+            return email_address, template_id, personalisation
 
-    return Message(
-        gettext(u"Your Civil Legal Advice reference number"),
-        recipients=[recipient],
-        body=render_template("emails/confirmation.txt", data=data),
-    )
+        # Decides between a personal callback or a third party callback
+        if data["callback"]["contact_number"]:
+            template_id = GOVUK_NOTIFY_TEMPLATES["PUBLIC_CALLBACK_WITH_NUMBER"]
+            personalisation.update(contact_number=data["callback"]["contact_number"])
+        elif data["thirdparty"]["contact_number"]:
+            template_id = GOVUK_NOTIFY_TEMPLATES["PUBLIC_CALLBACK_THIRD_PARTY"]
+            personalisation.update(contact_number=data["thirdparty"]["contact_number"])
+
+        return email_address, template_id, personalisation
+    except Exception as error:
```

Comment:
Why are we using a generic `Exception` here instead of explicit capturing exceptions such as `KeyError`?